### ISSUE TITLE
Try JUPnP

### ISSUE THREAD

**Issue by SubJunk** [UniversalMediaServer/UniversalMediaServer#2462]
Cling is unsupported and doesn't work well with JRE 11+. JUPnP has some fixes related to that, but still only claims to support JRE 8. It may be closer to what we want though, and because it is maintained, we can submit code changes to them instead of creating our own fork of Cling, which is what I've been doing.

This is a path to being able to fix bugs related to UPnP, so I am marking it as a bug instead of enhancement

**Comment by SubJunk** [comment]
I made an issue on JUPnP to ask questions and say hi https://github.com/jupnp/jupnp/issues/137

**Comment by valib** [comment]
@SubJunk made in #2464

**Comment by SubJunk** [comment]
@valib since it might take us a while to make JUPnP work with our implementation, maybe in the meantime we should go back to using Java 8, now that we know that Cling has compatibility problems with Java 9+. What do you think?

**Comment by SurfaceS** [comment]
I think changing Cling to jUPnP is not so hard to do, just few class to create / extends.
So if it's just replacing Cling by jUPnP, tell me I'll push something.

The fact is that UMS use only a small part of Cling, and Cling run the full service.
For now, it seems it is just used to discover the device/services from the network datagram messages.
Otherwise, only classes are used to build own message.
And UMS take care of datagram messages for serving... and http upnp.

Cling (or jUPnP) is a upnp server by itself, that expect to handle everything under it (datagram / http) with devices and services set in it and interfaced with it implementation.

**Comment by SubJunk** [comment]
@SurfaceS yes I think it would be good to change, because Cling is unsupported.
It would also mean that we are free to use the latest version of Maven, because the Cling 4thline repository holds us back

**Comment by SurfaceS** [comment]
Done on my side.
I'm waiting the #2777 commit to make a pull request for this, then I don't have to refactor again and again.
It add latest jUPnP dependency and remove Cling 4thline one.

I also have a media server RequestHandler V3, cleaned, and based only on java internal httpserver (no dependency).
It should work the same than V2 (that need netty dependency).

**Comment by SubJunk** [comment]
@SurfaceS great work, I'm reviewing #2777 now, sorry for the delay

**Comment by SubJunk** [comment]
@SurfaceS #2777 is merged now, looking forward to seeing that jupnp PR! I see the v3 one too which I'll look at now
